Modules/_json.c

```diff
@@ -1278,20 +1353,20 @@ encoder_call(PyEncoderObject *self, PyObject *args, PyObject *kwds)
     _PyUnicodeWriter_Init(&writer);
     writer.overallocate = 1;
 
-    PyObject *newline_indent = NULL;
+    PyObject *indent_cache = NULL;
     if (self->indent != Py_None) {
-        newline_indent = _create_newline_indent(self->indent, indent_level);
-        if (newline_indent == NULL) {
+        indent_cache = create_indent_cache(self, indent_level);
```

Comment:
When called via the public interface `indent_level` is always zero. By removing the argument `indent_level` in the internal implementation this call and some other bits of the code can be a bit simplified. 

(this is probably not worth creating a separate PR for, I leave it up to you)